core/eolearn/core/utils/parsing.py

```diff
@@ -267,13 +297,14 @@ def get_features(self, eopatch: Optional[EOPatch] = None) -> List[Tuple[FeatureT
 
         If `eopatch` is not provided the method fails if an all-feature request is in the specification.
         """
-        feature_names = []
-        for feature_type, name, _ in self._feature_specs:
-            self._validate_parsing_request(feature_type, name, eopatch)
-            if name is None and feature_type.has_dict():
-                feature_names.extend(list(zip(repeat(feature_type), eopatch[feature_type])))  # type: ignore
+        feature_names: List[Tuple[FeatureType, Optional[str]]] = []
+        for ftype, name, _ in self._feature_specs:
+            self._validate_parsing_request(ftype, name, eopatch)
+            if name is None and ftype.has_dict():
+                # _validate_parsing_request should ensure that eopatch is not None
+                feature_names.extend((ftype, name) for name in eopatch[ftype])  # type: ignore[index]
```

Comment:
Can you explain what is this `ignore[index]` ?

Reply:
the `_validate_parsing_request` should ensure that `eopatch` is not `None`. But `mypy` has no way of knowing that. So I have to ignore the case where `eopatch` is `None`.

Reply:
What if you do a hack that `_validate_parsing_request` accepts `Optional[EOPatch]` but then either returns an `EOPatch` or raises an error?

It's hacky, but that way the type changes and you wouldn't need to ignore it.

_this is just something that popped up into my mind, please note I'm not a typing expert_ 🙈

Reply:
i managed to throw it out in the new commit :D